lib/Analysis/Allocation.cpp

```diff
@@ -115,7 +115,12 @@ ScratchConfig getScratchConfigForCvt(RankedTensorType srcTy,
 
   assert(!isMfmaToDotShortcut(srcTy, dstTy));
 
-  auto [inOrd, outOrd] = getCvtOrder(srcLayout, dstLayout);
+  // FIXME This is NOT entirely correct
+  // This should be getElemOrder, but we don't have such a method
+  // TODO Implement getElemOrder and make sure it's consistent with
+  // getContigPerThread
+  auto inOrd = gpu::getThreadOrder(srcLayout);
```

Comment:
I think we assume `getElemOrder == getOrder`

Reply:
`getThreadOrder` is same as `getOrder` except for AMD's `AMDMfmaEncodingAttr`. I haven't taken a deep investigation.
pin @zhanglx13 for expertise maybe

Reply:
See that I changed the definition of `getThreadOrder` in this PR.

Reply:
To be specific I was referring to:
```
SmallVector<unsigned> AMDMfmaEncodingAttr::getThreadOrder() const {
  auto order = ::getOrder(*this);
  if (getIsTransposed())
    std::swap(order[0], order[1]);
  return order;
}
```
I'm not sure if we should use `getOrder` or `getThreadOrder` for this encoding